### PR TITLE
Tweaks and fixes for math built-ins

### PR DESCRIPTION
**Pull request by nmnobre**
Hi,

Simple and self-explanatory :)

Cheers,
-Nuno

### CODE REVIEW

File: include/hipSYCL/sycl/libkernel/sscp/builtins.hpp [hunk 1]
```diff
@@ -176,7 +176,7 @@ HIPSYCL_BUILTIN float __hipsycl_lgamma_r(float x, IntT* ptr) {
 }
 
 template<class IntT>
-HIPSYCL_BUILTIN double __hipsycl_lgamma_r_f64(double x, IntT* ptr) {
+HIPSYCL_BUILTIN double __hipsycl_lgamma_r(double x, IntT* ptr) {
```
Comment (illuhad) [inline include/hipSYCL/sycl/libkernel/sscp/builtins.hpp:179]: Wouldn't we also need a float overload that invokes `__hipsycl_sscp_lgamma_r_f32()`?

**Review by illuhad — COMMENTED** [review]
Awesome!

File: include/hipSYCL/sycl/libkernel/sscp/builtins.hpp (hunk 1 above)
Comment (nmnobre) [reply include/hipSYCL/sycl/libkernel/sscp/builtins.hpp:179]: It was already there, look a bit up.
I was just trying to fix a small typo/bug :)

Comment (illuhad) [reply include/hipSYCL/sycl/libkernel/sscp/builtins.hpp:179]: Ah, I got it. Sorry! :)

**Review by illuhad — APPROVED** [review]

**Pull request merged**